cli/lsp/language_server.rs

```diff
@@ -778,6 +819,210 @@ impl Inner {
+        LspError::invalid_request()
+      })?;
+
+    let cl = Rc::new(RefCell::new(Vec::new()));
+    navigation_tree.walk(&|i, mp| {
+      let mut code_lenses = cl.borrow_mut();
+
+      // TSC References Code Lens
+      if self.config.settings.enabled_code_lens_references() {
+        let source = CodeLensSource::References;
+        if let Some(parent) = &mp {
+          if parent.kind == tsc::ScriptElementKind::EnumElement {
+            code_lenses.push(i.to_code_lens(&line_index, &specifier, &source));
+          }
+        }
+        match i.kind {
+          tsc::ScriptElementKind::FunctionElement => {
+            if self
+              .config
+              .settings
+              .enabled_code_lens_references_all_functions()
+            {
+              code_lenses.push(i.to_code_lens(
+                &line_index,
+                &specifier,
+                &source,
+              ));
+            }
+          }
+          tsc::ScriptElementKind::ConstElement
+          | tsc::ScriptElementKind::LetElement
+          | tsc::ScriptElementKind::VariableElement => {
+            if EXPORT_MODIFIER.is_match(&i.kind_modifiers) {
+              code_lenses.push(i.to_code_lens(
+                &line_index,
+                &specifier,
+                &source,
+              ));
+            }
+          }
+          tsc::ScriptElementKind::ClassElement => {
+            if i.text != "<class>" {
+              code_lenses.push(i.to_code_lens(
+                &line_index,
+                &specifier,
+                &source,
+              ));
+            }
+          }
+          tsc::ScriptElementKind::InterfaceElement
+          | tsc::ScriptElementKind::TypeElement
+          | tsc::ScriptElementKind::EnumElement => {
+            code_lenses.push(i.to_code_lens(&line_index, &specifier, &source));
+          }
+          tsc::ScriptElementKind::LocalFunctionElement
+          | tsc::ScriptElementKind::MemberGetAccessorElement
+          | tsc::ScriptElementKind::MemberSetAccessorElement
+          | tsc::ScriptElementKind::ConstructorImplementationElement
+          | tsc::ScriptElementKind::MemberVariableElement => {
+            if let Some(parent) = &mp {
+              if parent.spans[0].start != i.spans[0].start {
+                match parent.kind {
+                  tsc::ScriptElementKind::ClassElement
+                  | tsc::ScriptElementKind::InterfaceElement
+                  | tsc::ScriptElementKind::TypeElement => {
+                    code_lenses.push(i.to_code_lens(
+                      &line_index,
+                      &specifier,
+                      &source,
+                    ));
+                  }
+                  _ => (),
+                }
+              }
+            }
+          }
+          _ => (),
+        }
+      }
+    });
+
+    self.performance.measure(mark);
+    Ok(Some(
+      Rc::try_unwrap(cl)
+        .map_err(|_| LspError::internal_error())?
```

Comment:
no need to handle the error gracefully, there should only be one reference here, use `unwrap()`